src/utils/compress_h5mu.py

```diff
@@ -59,3 +62,26 @@ def visit_path(
     with open(output_path, "br+") as f:
         nbytes = f.write(starting_metadata)
         f.write(b"\0" * (512 - nbytes))
+
+
+def write_h5ad_to_h5mu_with_compression(
+    output_file: Union[str, Path],
+    h5mu: Union[str, Path],
+    modality_name: str,
+    modality_data: AnnData,
+    output_compression=None,
+):
```

Comment:
in theory you could change the interface such that you can override multiple modalities without needing to shutil.copyfile the whole h5mu multiple times, e.g. with something like:

```
def write_h5ad_to_h5mu_with_compression(
    output_file: Union[str, Path],
    h5mu: Union[str, Path],
    new_modalities: Dict[str, AnnData],
    output_compression=None,
):
```

However, I don't think this is necessary per se. WDYT?

Reply:
Hmm, I can see where you are comming from, but I think if you want to write multiple modalities the modalities are already in memory? So a for a low-memory anndata to mudata writer is probably less useful in this usecase.